src/main/java/com/patreon/euphrates/S3Writer.java

```
@@ -64,7 +64,7 @@ public void shutdown() {
 
   protected void enqueueKey(Config.Table table, String key, ReusableCountLatch finished) {
     CopyJob job = new CopyJob(table, key, finished);
-    this.copyQueues.get(table.name).add(job);
+    this.copyQueues.get(table.name).offer(job);
```

So when this fails (and returns false) we drop the row on the floor...  Shouldn't we return the failure to the caller and let them retry/log/something?

---

Oh good catch. We probably should just increase the queue size instead and revert to the old crashing behaviour for now until someone can refactor this.